packages/remix-react/components.tsx

```diff
@@ -1090,11 +1096,20 @@ export function useSubmitImpl(key?: string): SubmitFunction {
         // <button>/<input type="submit"> may override attributes of <form>
 
         method =
-          options.method || target.getAttribute("formmethod") || form.method;
+          options.method ||
+          target.getAttribute("formmethod") ||
+          form.getAttribute("method") ||
+          defaultMethod;
         action =
-          options.action || target.getAttribute("formaction") || form.action;
+          options.action ||
+          target.getAttribute("formaction") ||
+          form.getAttribute("action") ||
```

Comment:
I think this needs to be resolved now relative to the route hierarchy like Link (not the URL segments the way the browser does it)

Reply:
We're not in a hook here so it's gonna be a bit tricky, will have to copy/pasta some stuff from here: https://github.com/remix-run/react-router/blob/main/packages/react-router/index.tsx#L611-L623

@mjackson why do the matches go through JSON.stringify and then JSON.parse in useResolvedPath?